module/os/linux/zfs/zfs_ctldir.c

```diff
@@ -1128,6 +1132,9 @@ zfsctl_snapshot_mount(struct path *path, int flags)
 	    zfsvfs->z_vfs->vfs_mntpoint ? zfsvfs->z_vfs->vfs_mntpoint : "",
 	    dname(dentry));
 
+	snprintf(options, 7, "%s",
+	    zfs_snapshot_no_setuid ? "nosuid" : "suid");
```

Comment:
Would it be better to have no option by default father than "suid"? I am not sure there is a difference, just wonder what if something else also try to control it by other means.

Reply:
No option is equivalent to "suid" and for the moment I think this keeps things more readable so let's keep it as is.